### DISCUSSION TITLE
Automatic Failover Test with only Two Nodes

### DISCUSSION THREAD

**Discussion by ahussainrc** [autobase-tech/autobase#660] [Q&A] [answered]
- I have setup Postgres Cluster using **scheme 1 PostgreSQL High-Availability only**
- I am using only two nodes node01 and node02, etcd is also on the same nodes. 
- I am also using VIP to have a single IP to connect to Postgres.
- node01 is the leader (master) and node02 is Replica
- To test the automatic failover, I shutdown the node02 with reboot command. 
- Automatic failover did happen actually and node02 became master but it only happened when node01 was started and up.  **Is it because the raft need to have more than half etcd servers to be up and running ?**
If yes, I will have to use 3 servers to have this automatic failover which I dont want. 

I have so many environments and having 3 nodes for each environment will consume too much resources. **Is there a way to achieve automatic failover with only two nodes ?**

**Comment by vitabaks** [comment, marked as answer]
You can deploy database clusters consisting of two servers, but there must be a DCS (etcd) cluster of at least three servers, here are the details http://thesecretlivesofdata.com/raft/

Create one etcd cluster on three separate small VMs and use it for multiple Postgres clusters. Variables `dcs_exists: true` and `patroni_etcd_hosts`

**Reply by ahussainrc** [reply]
Thanks @vitabaks for your response. 

Before I close the discussion, could you please give me an idea how much disk and memory would be consumed on etcd nodes for one postgres cluster.

**Reply by vitabaks** [reply]
Less than 1 gigabyte, because [auto compaction](https://github.com/vitabaks/postgresql_cluster/blob/master/roles/etcd/templates/etcd.conf.j2#L13) is executed.

**Reply by ahussainrc** [reply]
@vitabaks 
Thank you so much.
I really appreciate your quick help and support.
